components/exercises/chapter-4-oscar/Utils.tsx

```diff
@@ -70,3 +72,139 @@ export function reverseArrayInPlace(arr: any[]): undefined {
     // After the first iteration, the array becomes [5, 2, 3, 4, 1]
   }
 }
+
+/**
+ * Represents the properties of a list item.
+ */
+export type ListItemProp = {
+  value: ReactNode;
+  rest: ListItemProp | null;
+};
+
+/**
+ * Converts an array of numbers into a linked list.
+ * @param arr - The array of numbers to be converted.
+ * @returns The resulting linked list.
+ */
+export function arrayToList(arr: ReactNode[]): ListItemProp | null {
+  let list = null;
+
+  // arr.length - 1 --> because we want to start from the last element of the array as starts from 0
+  for (let i = arr.length - 1; i >= 0; i--) {
+    // list will store the result of every iteration, which starts on {value: 20, rest: null} so on
+    list = { value: arr[i], rest: list };
+  }
+  return list;
+}
+
+/**
+ * Converts a linked list to an array.
+ * @param list - The linked list to convert.
+ * @returns An array containing the values of the linked list.
+ */
+export function listToArray(list: ListItemProp | null): ReactNode[] {
+  let arr: ReactNode[] = [];
```

Comment:
doesn't have to be ReactNode[]

Reply:
same, doesn't belong, I will remove to avoid confusion